src/Service.Tests/Configuration/ConfigurationTests.cs

```diff
@@ -1331,6 +1331,177 @@ public async Task TestRuntimeBaseRouteInNextLinkForPaginatedRestResponse()
+                HttpResponseMessage followUpResponse = await client.SendAsync(followUpRequest);
+                Assert.AreEqual(HttpStatusCode.OK, followUpResponse.StatusCode);
+            }
+        }
+
+        /// <summary>
+        /// Validates the Location header field returned as part of POST requests
+        /// </summary>
+        /// <param name="entityType">Type of the entity</param>
+        /// <param name="requestPath">Request path for performing POST API requests on the entity</param>
+        /// <param name="baseRoute">Configured base route</param>
+        /// <param name="expectedLocationHeader">Expected value for Location field in the response header</param>
+        /// <returns></returns>
+        [DataTestMethod]
+        [TestCategory(TestCategory.MSSQL)]
+        [DataRow(EntitySourceType.Table, "/api/Book", "/data-api", "http://localhost/data-api/api/Book/id/")]
+        [DataRow(EntitySourceType.StoredProcedure, "/api/GetBooks", "/data-api", "http://localhost/data-api/api/GetBooks")]
+        public async Task ValidateLocationHeaderWhenBaseRouteIsConfigured(EntitySourceType entityType,
+                                                                          string requestPath,
+                                                                          string baseRoute,
+                                                                          string expectedLocationHeader)
+        {
+            GraphQLRuntimeOptions graphqlOptions = new(Enabled: true);
+            RestRuntimeOptions restRuntimeOptions = new(Enabled: true);
+
+            DataSource dataSource = new(DatabaseType.MSSQL,
+                GetConnectionStringFromEnvironmentConfig(environment: TestCategory.MSSQL), Options: null);
+
+            RuntimeConfig configuration;
+
+            if (entityType is EntitySourceType.StoredProcedure)
+            {
+                Entity entity = new(Source: new("get_books", EntitySourceType.StoredProcedure, null, null),
+                              Rest: new(new SupportedHttpVerb[] { SupportedHttpVerb.Get, SupportedHttpVerb.Post }),
+                              GraphQL: new(Singular: "GetBooks", Plural: "GetBooks"),
+                              Permissions: new[] { GetMinimalPermissionConfig(AuthorizationResolver.ROLE_ANONYMOUS) },
+                              Relationships: null,
+                              Mappings: null
+                             );
+
+                string entityName = "GetBooks";
+                configuration = InitMinimalRuntimeConfig(dataSource, graphqlOptions, restRuntimeOptions, entity, entityName);
+            }
+            else
+            {
+                configuration = InitMinimalRuntimeConfig(dataSource, graphqlOptions, restRuntimeOptions);
+            }
+
+            const string CUSTOM_CONFIG = "custom-config.json";
+
+            AuthenticationOptions AuthenticationOptions = new(Provider: EasyAuthType.StaticWebApps.ToString(), null);
```

Comment:
Can this be another DataRow in the test above? Looks like this test has lot of similar code than the previous...

Reply:
I started down that route but noticed that there were lots of branching logic needed in the tests.

Here are the places where branching logic was necessary.

a) Entity creation for tables vs stored procedures
b) Custom config file creation when base-route is configured vs not configured
c) Request body creation for tables vs stored-procedures 
d) Validations based on whether base-route is configured vs not configured.

Overall, having too many branching in a single test made the test cluttered. So, breaking them up into two different tests reduced the clutter and (hopefully) made it more intuitive to follow.